Framework/DataObjects/test/Histogram1DTest.h

```diff
@@ -45,6 +48,27 @@ class Histogram1DTest : public CxxTest::TestSuite {
     h2.setCountStandardDeviations(100);
   }
 
+  void test_copyDataFrom() {
+    Histogram1D histogram{Histogram::XMode::Points, Histogram::YMode::Counts};
+    histogram.setHistogram(Points(1), Counts(1));
+    EventList eventList;
+    eventList.setHistogram(BinEdges(2));
+    std::unique_ptr<const ISpectrum> specHist =
+        Kernel::make_unique<Histogram1D>(histogram);
+    std::unique_ptr<const ISpectrum> specEvent =
+        Kernel::make_unique<EventList>(eventList);
+    std::unique_ptr<ISpectrum> target = make_unique<Histogram1D>(
+        Histogram::XMode::Points, Histogram::YMode::Counts);
+
+    TS_ASSERT_THROWS_NOTHING(target->copyDataFrom(*specHist));
+    TS_ASSERT(target->points());
+    TS_ASSERT_EQUALS(&target->points()[0], &histogram.points()[0]);
+
+    TS_ASSERT_THROWS_NOTHING(target->copyDataFrom(*specEvent));
+    TS_ASSERT(target->binEdges());
+    TS_ASSERT_EQUALS(&target->binEdges()[0], &eventList.binEdges()[0]);
```

Comment:
Tests look ok. but minor point is there no `operator==` overload for `Histogram1D` that can be used rather than checking internal values?

Reply:
The test does not simply want to check equality but more: test that data is shared, so I am comparing addresses.